src/transformers/models/jukebox/modeling_jukebox.py

```diff
@@ -0,0 +1,2694 @@
+                self.encoder_dim = encoder_config.encoder_n_vocab
+                self.encoder = JukeboxConditionalAutoregressive(
+                    encoder_config,
+                    n_ctx=self.nb_relevant_lyric_tokens,
+                    embed_dim=self.encoder_dim,
+                    audio_conditioning=False,
+                    metadata_conditioning=False,
+                    is_encoder=True,
+                )
+                self.encoder.proj_in = JukeboxConv1D(encoder_config.width, config.width)
+                self.encoder.final_layer_norm = JukeboxLayerNorm(config.width)
+                self.encoder.lm_head = nn.Linear(config.width, encoder_config.encoder_n_vocab, bias=False)
+            else:
+                self.lyrics_enc_loss_dims = 0
+
+            # decoder model on the tokens
+            self.prior = JukeboxConditionalAutoregressive(
+                config,
+                audio_conditioning=(self.audio_conditioning or self.metadata_conditioning),
+                metadata_conditioning=self.metadata_conditioning,
+            )
+
+        self.next_token_prediction_loss_dims = config.n_ctx
+        self.total_loss_dims = self.lyrics_enc_loss_dims + self.next_token_prediction_loss_dims
+
+        self.downsamples = [stride**down for stride, down in zip(config.res_strides_t, config.res_downs_t)]
+        self.cond_downsample = self.downsamples[level] if level != 0 else None
+        self.raw_to_tokens = np.prod(self.downsamples[: nb_priors - level])
+        self.sample_length = self.n_ctx * self.raw_to_tokens
+
+        logger.info(
+            f"Level:{level}, Cond downsample:{self.cond_downsample}, Raw to tokens:{self.raw_to_tokens}, Sample"
+            f" length:{self.sample_length}"
+        )
+
+    def _get_encoder_config(self, config):
+        # Set config to use the lyric encoder parameters
+        encoder_config = copy.deepcopy(config)
+        encoder_config.attn_dropout = config.encoder_attn_dropout
+        encoder_config.attention_pattern = config.encoder_attention_pattern
+        encoder_config.blocks = config.encoder_blocks
+        encoder_config.depth = config.encoder_depth
+        encoder_config.emb_dropout = config.encoder_emb_dropout
+        encoder_config.heads = config.encoder_heads
+        encoder_config.init_scale = config.encoder_init_scale
+        encoder_config.loss_fraction = config.encoder_loss_fraction
+        encoder_config.attention_multiplier = config.encoder_attention_multiplier
+        encoder_config.mlp_multiplier = config.encoder_mlp_multiplier
+        encoder_config.resid_dropout = config.encoder_resid_dropout
+        encoder_config.attn_res_scale = config.encoder_attn_res_scale
+        encoder_config.spread = config.encoder_spread
+        encoder_config.width = config.encoder_width
+        encoder_config.zero_out = config.encoder_zero_out
+        encoder_config.n_vocab = config.encoder_n_vocab
+        return encoder_config
```

Comment:
Looks like it would be easier if the encoder config was nested as a config.

Reply:
Yes, but that would mean a nested config in a nested config of a nested config (`JukeboxConfig -> JukeboxPriorConfig -> JukeboxPriorConfig`) and thought it would be too heavy. Should I add it?